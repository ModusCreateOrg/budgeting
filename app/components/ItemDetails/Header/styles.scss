
@import 'theme/variables';

.outflow {
    color : $red;
}

.inflow {
    color : $green;
}