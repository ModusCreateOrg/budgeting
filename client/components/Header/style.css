a {
	padding: 35px 25px;
	margin-top: -6px;
	border-radius: 6px;
	font-size: 16px;
	color: #b1b1b1;
	border: 1px #dadada;
}

a:active, a.selected {
	color: black;
	text-decoration: underline;
	border: none;
	box-shadow: 0 1px 4px 1px rgba(0,0,0,0.16);
}
