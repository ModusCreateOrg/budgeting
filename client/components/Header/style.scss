@import 'theme/variables';

.header {
  position: relative;
  display: flex;
  margin-bottom: 3em;
}

.logo {
  width: 60px;
  height: 60px;
  margin: 10px 0px 0px 12px;
}

.gitHubButton {
  margin: 22px 14px 0px 0px;

  &:nth-of-type(1) {
    margin-left: auto;
  }
}

.navLink {
  padding: 35px 25px;
  margin: -6px 6px 0;
  border-radius: 6px;
  font-size: 16px;
  color: $gray;
  border: 1px solid $lightgray;
  text-decoration: none;

  &:active, &.selected {
    color: $black;
    text-decoration: underline;
    box-shadow: 0 1px 4px 1px $darkgray;
  }
}
