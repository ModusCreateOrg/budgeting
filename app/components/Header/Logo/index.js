// @flow
import * as React from 'react';
import logo from 'images/mclogo.svg';
import styles from './style.scss';

const Logo = () => (
  <a href="http://www.moduscreate.com">
    <img src={logo} alt="Modus Create" className={styles.logo} />
  </a>
);

export default Logo;
