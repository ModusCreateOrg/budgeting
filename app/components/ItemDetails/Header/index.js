import React from 'react';
import styles from './styles.scss';
import { ItemDetails } from 'modules/items';

const ItemDetailsHeader = ({ transaction: { description }, ratio, isInflow }: ItemDetails) => {
    const sign = isInflow ? "+" : "-";
    
    return (
        <div>
            <h1>Item details for {description}</h1>
            <h3><span className={isInflow ? styles.inflow : styles.outflow}>{sign}</span>{ratio}%</h3>
        </div>
    );
}

export default ItemDetailsHeader;