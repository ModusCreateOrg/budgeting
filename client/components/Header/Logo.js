import React from 'react';
import logo from 'images/mclogo.svg';
import styles from './style.scss';

const Logo = () => (
  <a className={styles.logo} href="http://www.moduscreate.com">
    <img src={logo} alt="Modus Create" />
  </a>
);

export default Logo;
