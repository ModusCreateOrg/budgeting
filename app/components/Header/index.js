// @flow
import * as React from 'react';

import NavLink from 'components/NavLink';
import GitHubButton from 'components/GitHubButton';
import Logo from './Logo';
import styles from './style.scss';

export default () => (
  <div className={styles.header}>
    <NavLink to="/budget" label="Budget" styles={styles} />
    <NavLink to="/reports" label="Reports" styles={styles} />
    <GitHubButton className={styles.gitHubButton} type="Star" />
    <GitHubButton className={styles.gitHubButton} type="Fork" />
    <Logo />
  </div>
);
