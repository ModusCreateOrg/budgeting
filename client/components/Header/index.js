import React, { PropTypes } from 'react';
import { Route, Link } from 'react-router-dom';
import Logo from './Logo';
import GitHubButton from '../GitHubButton';
import styles from './style.scss';

const NavLink = ({ to, label }) => (
  <Route
    to={to}
    exact
    children={({ location: { pathname } }) =>
      <Link to={to} className={`${styles.navLink} ${pathname === to ? styles.selected : ''}`}>{ label }</Link>
    }
  />
);

NavLink.propTypes = {
  to: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default () => (
  <div className={styles.header}>
    <NavLink to="/budget" label="Budget" />
    <NavLink to="/reports" label="Reports" />
    <GitHubButton className={styles.gitHubButton} type="Star" />
    <GitHubButton className={styles.gitHubButton} type="Fork" />
    <Logo />
  </div>
);
