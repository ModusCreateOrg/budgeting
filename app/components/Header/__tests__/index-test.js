import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../';

// mock nested components
jest.mock('components/NavLink', () => () => <div />);
jest.mock('components/GitHubButton', () => 'div');
jest.mock('../Logo', () => 'div');

it('renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
