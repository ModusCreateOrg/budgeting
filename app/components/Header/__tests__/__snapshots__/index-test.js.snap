// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header"
>
  <div />
  <div />
  <div
    className="gitHubButton"
    type="Star"
  />
  <div
    className="gitHubButton"
    type="Fork"
  />
  <div />
</div>
`;
